src/twisted/conch/ssh/transport.py

```diff
@@ -637,85 +690,152 @@ def sendPacket(self, messageType, payload):
             payload = self.outgoingCompression.compress(
                 payload
             ) + self.outgoingCompression.flush(2)
-        bs = self.currentEncryptions.encBlockSize
-        # 4 for the packet length and 1 for the padding length
-        totalSize = 5 + len(payload)
-        lenPad = bs - (totalSize % bs)
-        if lenPad < 4:
-            lenPad = lenPad + bs
-        packet = (
-            struct.pack("!LB", totalSize + lenPad - 4, lenPad)
-            + payload
-            + randbytes.secureRandom(lenPad)
+
+        wirePacket = self.currentEncryptions.createPacket(
+            payload,
+            self.outgoingPacketSequence,
         )
-        encPacket = self.currentEncryptions.encrypt(
-            packet
-        ) + self.currentEncryptions.makeMAC(self.outgoingPacketSequence, packet)
-        self.transport.write(encPacket)
+
+        self.transport.write(wirePacket)
         self.outgoingPacketSequence += 1
 
-    def getPacket(self):
+    def getPacket(self) -> bytes | None:
         """
         Try to return a decrypted, authenticated, and decompressed packet
         out of the buffer.  If there is not enough data, return None.
 
-        @rtype: L{str} or L{None}
-        @return: The decoded packet, if any.
+        @return: The decoded packet, if any. C{None} is returned if no packet was received yet.
         """
-        bs = self.currentEncryptions.decBlockSize
-        ms = self.currentEncryptions.verifyDigestSize
-        if len(self.buf) < bs:
+        if len(self.buf) < self.currentEncryptions.decBlockSize:
             # Not enough data for a block
             return
+
+        try:
+            if self.currentEncryptions.inMACType.endswith(b"-etm@openssh.com"):
```

Comment:
Maybe we can have something like:

```suggestion
            if self.currentEncryptions.isETM():
```